engine/src/witness/sol/nonce_witnessing.rs

```diff
@@ -11,16 +12,16 @@ use anyhow::{anyhow, Result};
 use serde_json::Value;
 use std::str::FromStr;
 
-pub async fn get_durable_nonces<SolRetryRpcClient>(
+pub async fn get_durable_nonce<SolRetryRpcClient>(
```

Comment:
I don't think I'd do these changes necessarily. We have an open ticket to get all durable nonces in one call so we don't have a separate `get_multiple_accounts` rpc call for each nonce but rather do them together. @kylezs

Reply:
It's effectively just a refactor in order to better reflect what we actually do currently, removing the need to eg. assert on the size of the returned vec. If we change it later, we can update the implementation to reflect that, but IMO the correct and most consistent way to structure this is to have one set of interfaces that reflect what we want to do from the voter's POV (in this case: get the nonce for a single account), and these should hide the details of the solana rpcs (use get_multiple with a single-value array).

Reply:
I agree that it's better with the current implementation, it's just that it will change when we implement the improved approach. But it's a fairly easy change so it can be done if/when we do that as part of the ticket :+1: